test/e2e/bundle_e2e_test.go

```diff
@@ -165,6 +165,87 @@ var _ = Describe("Installing bundles with new object types", func() {
 			}).Should(Succeed())
 		})
 	})
+
+	When("A bundle is installed with a CR that is associated with a provided CRD", func() {
```

Comment:
Does this test always hit the race condition?

Reply:
yes I believe so

A similar test failed like 99/100 times in the first gardener PR. It depends on the api-server a bit but basically yes it would fail consistently without the code change.